### ISSUE TITLE
problem while deleting

### ISSUE THREAD

**Issue by Xebache** [d-markey/azure_cosmosdb#1]
Hi!

I am using this connector in a project and i have an issue when i delete an item from cosmosDb : "BadResponseException: Unsupported content-type: null"; i just don't know where it comes from...
The code i use in flutter is pretty straight forward : 

```
Future<Result> delete(String id) async {
    try {
      await CosmosService().entityCollection.delete(id);
      return Result.success();
    } catch (e) {
      logger.wtf(e.toString());
      return Result.fail();
    }
  }

```

The item is actually deleted from cosmosDb but i always fall in the catch part... i tried to print the boolean result expected from the delete method but never had a chance to see it... 

All the other CRUDL methods work great but this one is resisting... Can you please tell me if i am doing something wrong or if there is something on your side that could help me?

Thanks!

**Comment by d-markey** [comment]
Hello @Xebache,

I will need some more context to help you with this. I can trace the origin in `_client.dart`:

```dart
    if (contentType != MimeType.json) {
      throw BadResponseException('Unsupported content-type: $contentType');
    }
```

I'd be interested in checking the HTTP request/response. You can enable a debug HTTP client by importing `azure_cosmosdb_debug.dart` instead of `azure_cosmosdb.dart`. Then you can provide a debug HTTP client when connecting with the database :

```dart
  final server = CosmosDbServer(
    'https://localhost:8081',
    masterKey: masterKey,
    httpClient: DebugHttpClient(),
  );
  server.useLogger(print, withBody: true, withHeader: true);
```

Please locate the `DELETE` command in the log and post the request/response here so I can investigate.

Make sure you don't advertise your master key or any other sensitive data when posting this material ;-)

Regards

**Comment by d-markey** [comment]
BTW I'm working on a version 2. It will have some breaking changes but also provide improvements like batch processing and (experimental) support of `MultiHash` partition keys and partition keys in general (at least in my opinion). I'll hopefully post something this week-end.

**Comment by Xebache** [comment]
Hi @d-markey !

here is what i get in the log:

>>>>>>>>>>>>>>>
[2023-01-06T22:35:31.518Z] --> DELETE https://.../docs/2pyLxW50iuNtYz6B4vciBEwmiOh1
[2023-01-06T22:35:31.518Z] --> authorization = ...
[2023-01-06T22:35:31.518Z] --> x-ms-date = Fri, 06 Jan 2023 22:35:31 GMT
[2023-01-06T22:35:31.518Z] --> x-ms-version = 2018-12-31
[2023-01-06T22:35:31.518Z] --> user-agent = AzureCosmosDb.Dart/1.8.0
[2023-01-06T22:35:31.518Z] --> x-ms-documentdb-partitionkey = ["2pyLxW50iuNtYz6B4vciBEwmiOh1"]
[2023-01-06T22:35:31.518Z] --> body =
[2023-01-06T22:35:31.518Z] <-- COSMOS DB Status = 204
[2023-01-06T22:35:31.518Z] <-- access-control-allow-credentials = true
[2023-01-06T22:35:31.518Z] <-- access-control-allow-origin = http://localhost:8000
[2023-01-06T22:35:31.518Z] <-- cache-control = no-store, no-cache
[2023-01-06T22:35:31.518Z] <-- content-length = 0
[2023-01-06T22:35:31.518Z] <-- pragma = no-cache
[2023-01-06T22:35:31.518Z] <-- strict-transport-security = max-age=31536000
[2023-01-06T22:35:31.518Z] <-- x-ms-activity-id = 3e9aa140-c981-4e9c-95b4-507ae5f86822
[2023-01-06T22:35:31.518Z] <-- COSMOS DB 204 Payload =
<<<<<<<<<<<<<<<<
👾  UserService - DELETE dbs/.../colls/Users/docs/2pyLxW50iuNtYz6B4vciBEwmiOh1: BadResponseException: Unsupported content-type: null

i also noticed in the browser's inspector that the request has no response data available... but the item is deleted from the db.

Another matter, not related at all: i have a list of 205 items that i have to import at once; i had to import your connector locally to add Paging to the Context and allow to specify a max length for the list() method. It would be nice to allow user to fetch more than 100 items if needed :-))

**Comment by d-markey** [comment]
Hello, I've just published version 1.8.1 which should fix your problem. Let me know if everything's OK now!

**Comment by Xebache** [comment]
Yup, it worked! Thank you !!!
